designs/2019-rule-tester-improvements/design.md

```diff
@@ -0,0 +1,83 @@
+- Start Date: 2019-06-07
+- RFC PR: https://github.com/eslint/rfcs/pull/25
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# `RuleTester` Improvements
+
+## Summary
+
+This RFC improves `RuleTester` to check more mistakes.
+
+## Motivation
+
+`RuleTester` overlooks some mistakes.
+
+- Using non-standard properties of AST ([typescript-eslint/typescript-eslint#405](https://github.com/typescript-eslint/typescript-eslint/issues/405)).<br>
```

Comment:
How can this be checked for arbitrary parsers? It’s both common and imo a best practice for a rule to declare visitors for the union of parse nodes it might encounter - including flow or TS or babel nodes that eslint might not know about.

Reply:
This RFC addresses only `node.start` and `node.end` because using `start`/`end` has often caused problems. (the standard property is `node.range`.)